src/main/java/org/htmlunit/javascript/host/URLSearchParams.java

```diff
@@ -137,26 +143,162 @@ public URLSearchParams() {
      */
     @JsxConstructor
     public URLSearchParams(final Object params) {
-        // TODO: Pass in a sequence
-        // new URLSearchParams([["foo", 1],["bar", 2]]);
-
-        // TODO: Pass in a record
-        // new URLSearchParams({"foo" : 1 , "bar" : 2});
-
         url_ = new URL("http://www.htmlunit.org", "");
 
         if (params == null || Undefined.isUndefined(params)) {
             return;
         }
 
+        // check if params is a sequence
+        String paramsString = handleSequence(params);
+        if (paramsString == null) {
+            // if params is a record
+            if (params instanceof NativeObject) {
+                StringBuilder sb = new StringBuilder();
+                for (Map.Entry<Object, Object> keyValuePair : ((NativeObject)params).entrySet()) {
+                    if (sb.length() > 0) {
+                        sb.append('&');
+                    }
+
+                    sb.append(Context.toString(keyValuePair.getKey()))
+                            .append('=')
+                            .append(Context.toString(keyValuePair.getValue()));
+                }
+                paramsString = sb.toString();
+            } else {
+                // otherwise process it as a string
+                paramsString = Context.toString(params);
+            }
+        }
+
         try {
-            url_.setSearch(splitQuery(Context.toString(params)));
+            url_.setSearch(splitQuery(paramsString));
         }
         catch (final MalformedURLException e) {
             LOG.error(e.getMessage(), e);
         }
     }
 
+    private String handleSequence(Object sequence) {
```

Comment:
instead of this ad-hoc "return null" interface, could you just put everything inside this
```suggestion
    private static String createParamString(final Object paramArg) {
```

Reply:
did you forget to push?